etl/src/main/java/org/pih/hivmigration/etl/sql/EncounterMigrator.groovy

```diff
@@ -54,16 +55,21 @@ class EncounterMigrator extends SqlMigrator {
                     e.performed_by,
                     e.encounter_site,
                     e.note_title,
-                    e.response_to
+                    e.response_to,
+                    NVL(i.FORM_VERSION,f.FORM_VERSION)
                 from
-                    hiv_encounters e, hiv_demographics_real p 
-                    where e.patient_id = p.patient_id 
+                    hiv_encounters e, hiv_demographics_real p, HIV_INTAKE_FORMS i, HIV_FOLLOWUP_FORMS f
+                    where e.patient_id = p.patient_id and i.ENCOUNTER_ID (+)= e.ENCOUNTER_ID and f.ENCOUNTER_ID (+) = e.ENCOUNTER_ID
             '''
         )
         executeMysql("Add UUIDs", '''
             UPDATE hivmigration_encounters SET encounter_uuid = uuid();
         ''')
 
+        executeMysql("Set Form Version to 1 for all Intake and Followup with no versions", '''
+            update hivmigration_encounters set form_version='1' where (source_encounter_type='intake' or source_encounter_type='followup') and form_version is null;
+        ''')
```

Comment:
You could use the coalesce function instead of the nvl function above to achieve this same thing I believe - eg. coalesce(i.form_version, f.form_version, 1) as form_version.  Just FYI.

Reply:
Oh, right, I may try that, thanks!